action.yml

```
@@ -1,12 +1,212 @@
+          exit 1
+        fi
+        # Check if there is y.2.0 tag for the current supported major version of framework
+        FRAMEWORK_TWO_TAG_EXISTS=0
+        TWO_TAG=$(jq -r --arg framework_supported_major "$FRAMEWORK_SUPPORTED_MAJOR" '.[] | select(.name == $framework_supported_major + ".2.0")' __tags.json)
+        if ! [[ -z $TWO_TAG ]]; then
+          FRAMEWORK_TWO_TAG_EXISTS=1
+        fi
+        echo "FRAMEWORK_TWO_TAG_EXISTS is $FRAMEWORK_TWO_TAG_EXISTS"
+        echo "framework_two_tag_exists=$FRAMEWORK_TWO_TAG_EXISTS" >> "$GITHUB_OUTPUT"
+        # Check if there is y.4.0 tag for the current supported major version of framework
+        FRAMEWORK_FOUR_TAG_EXISTS=0
+        FOUR_TAG=$(jq -r --arg framework_supported_major "$FRAMEWORK_SUPPORTED_MAJOR" '.[] | select(.name == $framework_supported_major + ".4.0")' __tags.json)
+        if ! [[ -z $FOUR_TAG ]]; then
+          FRAMEWORK_FOUR_TAG_EXISTS=1
+        fi
+        echo "FRAMEWORK_FOUR_TAG_EXISTS is $FRAMEWORK_FOUR_TAG_EXISTS"
+        echo "framework_four_tag_exists=$FRAMEWORK_FOUR_TAG_EXISTS" >> "$GITHUB_OUTPUT"
+        rm __tags.json
+
+    - name: Derive module branch
+      id: derive-module-branch
+      shell: bash
+      env:
+        GITHUB_REPOSITORY: ${{ github.repository }}
+        DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+        BRANCH_TYPE: ${{ steps.derive-branch-type.outputs.branch_type }}
+        FRAMEWORK_SUPPORTED_MAJOR: ${{ steps.derive-supported-framwork.outputs.framework_supported_major }}
+        FRAMEWORK_TWO_TAG_EXISTS: ${{ steps.derive-supported-framwork.outputs.framework_two_tag_exists }}
+        FRAMEWORK_FOUR_TAG_EXISTS: ${{ steps.derive-supported-framwork.outputs.framework_four_tag_exists }}
+      run: |
+        # $DEFAULT_BRANCH is always assumed to be a numeric value i.e. not "main", and is assumed to be the same CMS version as $FRAMEWORK_SUPPORTED_MAJOR
+        # It's always assumed that this action is run every six months, and at least one month from a minor beta release
+        # meaning that whenever we get the highest minor branch for the current minor, that minor branch will currently be stable
+        echo "DEFAULT_BRANCH is $DEFAULT_BRANCH"
+        MAJOR=""
+        if [[ "$BRANCH_TYPE" == "prev-major-curr-minor" ]]; then
+          if [[ "$FRAMEWORK_TWO_TAG_EXISTS" == "1" ]]; then
+            echo "Because framework tag $FRAMEWORK_SUPPORTED_MAJOR.2.0 exists, the 'prev-major-curr-minor' run type is no longer supported"
+            exit 0
+          fi
+          MAJOR=$((DEFAULT_BRANCH - 1))
+        elif [[ "$BRANCH_TYPE" == "curr-major-curr-minor" ]]; then
+          MAJOR="$DEFAULT_BRANCH"
+        elif [[ "$BRANCH_TYPE" == "curr-major-next-minor" ]]; then
+          if [[ "$FRAMEWORK_FOUR_TAG_EXISTS" == "1" ]]; then
+            echo "Because framework tag $FRAMEWORK_SUPPORTED_MAJOR.4.0 exists, the 'curr-major-next-minor' run type should not create create a pull-request"
```

```suggestion
            echo "The current major release line will have no further minor releases, so the 'curr-major-next-minor' run type should not create a pull-request"
```
This means we don't have to mentally translate "x.4.0 exists" into "no next minor" because it's just spelled out directly.

---

Updated